### PR TITLE
allow datafusion-cli to take a file param

### LINKED ISSUE #282: datafusion-cli to take a file as a command input
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
(This section helps Arrow developers understand the context and *why* for this feature, in addition to  the *what*)

A challenge to add `-f` param to `datafusion-cli` so that instead of being interactive shell it can execute commands from a file.

**Describe the solution you'd like**

like the `psql` command that takes a `-f` param.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

Closes #282.

Relies on #283 

 # Rationale for this change

Allow datafusion-cli to take file as a param.

# What changes are included in this PR?

See above

# Are there any user-facing changes?

User facing changes are the sole addition of such flag.

### CODE REVIEW

**Comment by Jimexist** [comment]
need to fix https://github.com/apache/arrow-datafusion/pull/286 first

**Comment by Jimexist** [comment]
need to fix https://github.com/apache/arrow-datafusion/pull/286 first

File: datafusion-cli/src/main.rs [hunk 1]
```diff
@@ -65,16 +77,91 @@ pub async fn main() {
         execution_config = execution_config.with_batch_size(batch_size);
     };
 
-    let mut ctx =
-        ExecutionContext::with_config(execution_config.with_information_schema(true));
+    if let Some(file) = matches
+        .value_of("file")
+        .and_then(|file| File::open(&file).ok())
```
Comment (alamb) [inline datafusion-cli/src/main.rs:82]: I think the use of `ok()` silently ignore errors when opening the file and silently fallback to the repl. This leads to strange behavior for certain files, such as those that that exists, but I don't have permissions to read

For a file such as this, I would expect `datafusion-cli` to generate an error about permissions:

```
(arrow_dev) alamb@ip-10-0-0-124:~/Software/arrow-datafusion$ ls -l /tmp/foo.sql 
--w-------  1 alamb  wheel     0B May  7 17:30 /tmp/foo.sql

(arrow_dev) alamb@ip-10-0-0-124:~/Software/arrow-datafusion$ cat /tmp/foo.sql 
cat: /tmp/foo.sql: Permission denied
```

However, the code on this branch generates no error and instead drops into the interactive REPL loop

```
(arrow_dev) alamb@ip-10-0-0-124:~/Software/arrow-datafusion$ ./target/debug/datafusion-cli --file=/tmp/foo.sql
> 

```

**Review by alamb — COMMENTED** [review]
Thanks @Jimexist ! This is a great feature to add. 

I think there is a bug in error handling for files, but otherwise this looks great!

File: datafusion-cli/src/main.rs (hunk 1 above)
Comment (Jimexist) [reply datafusion-cli/src/main.rs]: @alamb thanks for the comment, it's fixed now

**Comment by alamb** [comment]
Looks really good -- thanks @Jimexist

**Review by alamb — APPROVED** [review]

**Pull request merged**